File: 61_Rotate_List/solution.c
struct ListNode* rotateRight(struct ListNode* head, int k) {
    if (!head || !head->next || k == 0) return head;
    int l = 1;
    struct ListNode* t = head;
    while (t->next) {
        t = t->next;
        l++;
    }
    k %= l;
    if (k == 0) return head;

    t->next = head;
    int s = l - k;
    struct ListNode* nt = head;
    for (int i = 1; i < s; i++) nt = nt->next;

    struct ListNode* nh = nt->next;
    nt->next = NULL;
    return nh;
}

File: 61_Rotate_List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or not head.next or k == 0:
            return head

        l = 1
        t = head
        while t.next:
            t = t.next
            l += 1

        k %= l
        if k == 0:
            return head

        t.next = head

        s = l - k
        nt = head
        for _ in range(1, s):
            nt = nt.next

        nh = nt.next
        nt.next = None

        return nh
